WalletWasabi.Fluent/ViewModels/Wallets/Send/PrivacySuggestionsFlyoutViewModel.cs

```diff
@@ -82,10 +84,13 @@ public async Task BuildPrivacySuggestionsAsync(Wallet wallet, TransactionInfo in
 					// Only allow to create 1 more input with BnB. This accounts for the change created.
 					int maxInputCount = transaction.SpentCoins.Count() + 1;
 
+					var cjManager = Services.HostedServices.Get<CoinJoinManager>();
+					ImmutableList<SmartCoin> coinsToExclude = cjManager.CoinsInCriticalPhase[wallet.WalletName];
+
 					var pockets = wallet.GetPockets();
 					var spentCoins = transaction.SpentCoins;
 					var usedPockets = pockets.Where(x => x.Coins.Any(coin => spentCoins.Contains(coin)));
-					var coinsToUse = usedPockets.SelectMany(x => x.Coins).ToImmutableArray();
+					var coinsToUse = usedPockets.SelectMany(x => x.Coins).Except(coinsToExclude).ToImmutableArray();
```

Comment:
An edge case thought but if the original txn cannot avoid the cjing coins, here they will be ignored for sure so the change avoidance feature will be unavailable while it might produce some options.

You could check if `spentCoins` contains any coin from `coinsToExclude` (which means the original transaction couldn't avoid the CJing coins), if so then do not remove the `coinsToExclude` coins from `coinsToUse` otherwise do it.